sp/src/game/server/ez2/npc_flyingpredator.cpp

```diff
@@ -201,9 +297,29 @@ void CNPC_FlyingPredator::Precache()
 //=========================================================
 int CNPC_FlyingPredator::RangeAttack1Conditions( float flDot, float flDist )
 {
-	if ( flDist <= m_flDistTooFar && flDot >= 0.5 && gpGlobals->curtime >= m_flNextSpitTime)
+	if ( flDist <= InnateRange1MaxRange() && gpGlobals->curtime >= m_flNextSpitTime)
 	{
-		m_flNextSpitTime = gpGlobals->curtime + GetMinSpitWaitTime();
+		if ( CapableOfFlyNav() )
+		{
+			if (flDist <= InnateRange1MinRange())
```

Comment:
Why is this case specifically for when the stukabat has the capability to use flying navigation?

Reply:
IIRC this is related to flying-capable stukabats being able to go directly back to flying after dive bombing at certain heights. This condition prevents them from getting caught in a loop. Non-flying stukabats don't have this problem because they always fall to the ground after a dive bomb.